Refactor OS matrix to support different architectures

rnetser opened the pull request:
##### Short description:

##### More details:

##### What this PR does / why we need it:

##### Which issue(s) this PR fixes:

##### Special notes for reviewer:

##### jira-ticket:
<!--  full-ticket-url needs to be provided. This would add a link to the pull request to the jira and close it when the pull request is merged
If the task is not tracked by a Jira ticket, just write "NONE".
-->


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Added support for `arm64` and `s390x` architectures (s390x as work in progress) in test configuration and documentation.
  - Introduced new configuration files and utilities for architecture-specific OS image matrices.
  - Added new architecture-specific image classes and constants for test images.

- **Bug Fixes**
  - Improved robustness of dictionary access throughout tests and utilities to prevent errors when configuration keys are missing.

- **Refactor**
  - Consolidated and modularized OS image and template configuration for easier architecture extension.
  - Updated test and utility code to use safer methods for dictionary value retrieval and path construction.
  - Replaced platform-dependent path joins with consistent string formatting for image paths.
  - Enhanced pytest matrix parameter handling to avoid errors during collection or setup phases.

- **Documentation**
  - Added detailed architecture support documentation in a new file.
  - Updated getting started guide to reflect multi-architecture support and configuration options.

- **Chores**
  - Updated test and tox configuration to support architecture-specific test runs and environments.
  - Removed default test config file from pytest options to allow flexible configuration loading.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

cnv-qe-bot1 commented:
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
    * You can add extra args to the Podman build command
        * Example: `/build-and-push-container --build-arg OPENSHIFT_PYTHON_WRAPPER_COMMIT=<commit_hash>`
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`
 * to assign reviewer to PR use `/assign-reviewer @<reviewer>`

PR will be approved when the following conditions are met:
 * `/approve` from one of the approvers.
 * Minimum number of required `/lgtm` (`2`) is met.

<details>
<summary>Approvers and Reviewers</summary>


 * Approvers:
   * dshchedr
   * dshchedr
   * myakove
   * rnetser
   * rnetser
   * vsibirsk
   * vsibirsk


 * Reviewers:
   * RoniKishner
   * RoniKishner
   * dshchedr
   * dshchedr
   * geetikakay
   * rnetser
   * rnetser
   * vsibirsk
   * vsibirsk


</details>

<details>
<summary>Supported /retest check runs</summary>

 * `/retest tox`: Retest tox
 * `/retest build-container`: Retest build-container
 * `/retest all`: Retest all

</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm
 * approve

</details>

polarion-jenkins reviewed (changes requested):
D/S test `tox -e verify-bugs-are-open` failed: cnv-tests-tox-executor/11098

vsibirsk commented on tests/global_config_x86.py:
```diff
@@ -0,0 +1,78 @@
+
+centos_os_matrix = generate_os_matrix_dict(os_name="centos", supported_operating_systems=["centos-stream-9"])
+
+instance_type_rhel_os_matrix = [
+    {
+        "rhel-10": {
+            OS_VERSION_STR: "10",
+            DV_SIZE_STR: Images.Rhel.DEFAULT_DV_SIZE,
+            INSTANCE_TYPE_STR: "u1.medium",
+            PREFERENCE_STR: "rhel.10",
+            DATA_SOURCE_NAME: "rhel10",
+            LATEST_RELEASE_STR: True,
+        }
+    },
+]
```
should remain in `global_config.py` as it's not arch related

rnetser replied:
currently only supported for x86

rnetser replied:
done; moved back

vsibirsk commented on utilities/constants.py:
```diff
@@ -110,10 +110,53 @@ class Cdi:
+
+        class Rhel:
+            RHEL9_5_IMG = "rhel-95-s390x.qcow2"
+            RHEL9_5_ARM64_IMG = "rhel-95-aarch64.qcow2"
```
this is arm64 image. need to create and move it unget Arm64 class (can be done in separate pr)

rnetser replied:
done

vsibirsk commented on utilities/constants.py:
```diff
@@ -110,10 +110,53 @@ class Cdi:
-    arch = os.environ.get("OPENSHIFT_VIRTUALIZATION_TEST_IMAGES_ARCH", "x86_64")
-    if arch not in ("x86_64",):
+    arch = os.environ.get("OPENSHIFT_VIRTUALIZATION_TEST_IMAGES_ARCH", X86_64)
+    if arch not in (X86_64, S390X):
```
any reason to do it upper case?

rnetser replied:
using existing constants

vsibirsk replied:
ops. my bad. i missed that it's constants now...

polarion-jenkins reviewed (changes requested):
D/S test `tox -e verify-bugs-are-open` failed: cnv-tests-tox-executor/11100

polarion-jenkins reviewed (changes requested):
D/S test `tox -e verify-bugs-are-open` failed: cnv-tests-tox-executor/11114

rnetser commented:
/retest tox

rnetser commented on utilities/constants.py:
```diff
@@ -110,10 +110,53 @@ class Cdi:
+            RHEL10_REGISTRY_GUEST_IMG = "registry.redhat.io/rhel10-beta/rhel-guest-image"
+            DIR = f"{BASE_IMAGES_DIR}/rhel-images"
+            DEFAULT_DV_SIZE = "20Gi"
+            DEFAULT_MEMORY_SIZE = "1.5Gi"
```
`RHEL9_5_ARM64_IMG` was removed in new code

rnetser commented on utilities/constants.py:
```diff
@@ -110,10 +110,53 @@ class Cdi:
+            DEFAULT_DV_SIZE = "10Gi"
+
+
+# Architectures
+X86_64 = "x86_64"
+AMD_64 = "amd64"
+ARM_64 = "arm64"
+S390X = "s390x"
+
```
`ARM_64` was added in new commit

rnetser replied:
in the current contex AMD_64 and AMD_64 are the same in terms of the tests

rnetser replied:
moved to upper case to have one way in anming the classes

rnetser commented on utilities/constants.py:
```diff
@@ -110,10 +110,53 @@ class Cdi:
+            DISK_DEMO = "fedora-cloud-registry-disk-demo"
+            DIR = f"{BASE_IMAGES_DIR}/fedora-images"
+            DEFAULT_DV_SIZE = "10Gi"
+            DEFAULT_MEMORY_SIZE = "1Gi"
```
added a todo comment on cirros; this is a workaround

polarion-jenkins reviewed (changes requested):
D/S test `tox -e verify-bugs-are-open` failed: cnv-tests-tox-executor/11120

rnetser commented on utilities/os_utils.py:
```diff
@@ -0,0 +1,198 @@
+    class_name = "CentOS" if os_name == "centos" else os_name.title()
+    os_base_class = getattr(Images, class_name)
+
+    os_formatted_list = []
```
Added `AttributeError` catching

rnetser replied:
please check updated code

rnetser commented on tests/global_config_x86.py:
```diff
@@ -0,0 +1,78 @@
+    latest_centos_os_dict,
+) = get_latest_os_dict_list(os_list=[rhel_os_matrix, windows_os_matrix, fedora_os_matrix, centos_os_matrix])
+
+for _dir in dir():
+    if not config:  # noqa: F821
+        config: dict[str, Any] = {}
+    val = locals()[_dir]
```
this is how global_config works

polarion-jenkins reviewed (changes requested):
D/S test `tox -e verify-bugs-are-open` failed: cnv-tests-tox-executor/11124

rnetser commented on tests/global_config_x86.py:
```diff
@@ -0,0 +1,78 @@
+from utilities.infra import get_latest_os_dict_list
+from utilities.os_utils import generate_os_matrix_dict
+
+global config
+global_config = pytest_testconfig.load_python(py_file="tests/global_config.py", encoding="utf-8")
```
this soltion does not work, resulting config does not have the values from this file

rnetser commented:
@coderabbitai review

rnetser commented:
/retest tox

rnetser commented on tests/global_config_oci.py:
```diff
@@ -13,7 +13,7 @@
 
 global config
-global_config = pytest_testconfig.load_python(py_file="tests/global_config.py", encoding="utf-8")
+global_config = pytest_testconfig.load_python(py_file="testsglobal_config_x86.py", encoding="utf-8")
```
fixed

jpeimer commented on conftest.py:
```diff
@@ -311,7 +311,7 @@ def add_polarion_parameters_to_user_properties(item: Item, matrix_name: str) ->
     for value in values:
         value = value.strip("#")
-        for param in py_config[matrix_name]:
+        for param in py_config.get(matrix_name, {}):
```
If there's no `py_config.get(matrix_name)`, then we don't need to loop over `values`, right? We can check the matrix at the beginning?

rnetser replied:
done

jpeimer commented on libs/infra/images.py:
```diff
@@ -0,0 +1,97 @@
+    RHEL9_6_IMG: str | None = None
+    RHEL8_REGISTRY_GUEST_IMG: str = "registry.redhat.io/rhel8/rhel-guest-image"
+    RHEL9_REGISTRY_GUEST_IMG: str = "registry.redhat.io/rhel9/rhel-guest-image"
+    # TODO: change back to registry.redhat.io when rhel10 is available
+    RHEL10_REGISTRY_GUEST_IMG: str = "registry.stage.redhat.io/rhel10/rhel-guest-image"
```
It should be available now

rnetser replied:
should be updated and verified in a separate PR

jpeimer commented on utilities/os_utils.py:
```diff
@@ -0,0 +1,222 @@
+    elif os_name == "centos":
+        base_dict = CENTOS_OS_MAPPING
+    else:
+        raise ValueError(f"Unsupported OS: {os_name}. Supported: rhel, win, fedora")
```
```suggestion
        raise ValueError(f"Unsupported OS: {os_name}. Supported: rhel, win, fedora, centos")
```

rnetser replied:
done

rnetser commented on tests/conftest.py:
```diff
@@ -1881,12 +1881,14 @@ def rhel_latest_os_params():
+            "rhel_template_labels": latest_rhel_dict["template_labels"],
+        }
+
+    return {}
```
should we? @vsibirsk ?

rnetser commented on tests/storage/cdi_clone/test_clone.py:
```diff
@@ -157,9 +155,9 @@ def test_successful_vm_restart_with_cloned_dv(
             {
-                "vm_name": f"vm-win-{WINDOWS_11['os_version']}",
+                "vm_name": f"vm-win-{WINDOWS_11.get('os_version')}",
                 "template_labels": WINDOWS_11_TEMPLATE_LABELS,
-                "os_version": WINDOWS_11["os_version"],
+                "os_version": WINDOWS_11.get("os_version"),
```
see above comment on this is done for collection only; during test execution, tests will not be executed if not part of the architecture and have windows set in pyconfig

rnetser commented on tests/storage/memory_dump/test_memory_dump.py:
```diff
@@ -16,9 +16,9 @@
                 "dv_name": "dv-windows",
-                "image": WINDOWS_LATEST["image_path"],
+                "image": WINDOWS_LATEST.get("image_path"),
                 "storage_class": py_config["default_storage_class"],
-                "dv_size": WINDOWS_LATEST["dv_size"],
+                "dv_size": WINDOWS_LATEST.get("dv_size"),
```
this is done for collection only; during test execution, tests will not be executed if not part of the architecture and have windows set in pyconfig

rnetser commented on tests/virt/cluster/sysprep/test_sysprep.py:
```diff
@@ -253,8 +253,8 @@ def detached_sysprep_resource_and_restarted_vm(sysprep_vm, attached_sysprep_volu
                 "dv_name": WINDOWS_2019_OS,
-                "image": WINDOWS_2019["image_path"],
-                "dv_size": WINDOWS_2019["dv_size"],
+                "image": WINDOWS_2019.get("image_path"),
+                "dv_size": WINDOWS_2019.get("dv_size"),
```
this is done for collection only; during test execution, tests will not be executed if not part of the architecture and have windows set in pyconfig

rnetser commented on tests/virt/node/general/test_networkinterfacemultiqueue_feature.py:
```diff
@@ -169,8 +167,8 @@ def test_two_cores_two_sockets_two_threads(self, network_interface_multiqueue_vm
-                "image": os.path.join(Images.Windows.HA_DIR, Images.Windows.WIN2k19_HA_IMG),
-                "dv_size": WINDOWS_2019["dv_size"],
+                "image": f"{Images.Windows.HA_DIR}/{Images.Windows.WIN2k19_HA_IMG}",
+                "dv_size": WINDOWS_2019.get("dv_size"),
```
this is done for collection only; during test execution, tests will not be executed if not part of the architecture and have windows set in pyconfig

rnetser commented on tests/virt/node/hyperv_support/test_hyperv_features_in_vm.py:
```diff
@@ -62,8 +62,8 @@ def verify_evmcs_related_attributes(vmi_xml_dict):
                 "dv_name": WINDOWS_LATEST_OS,
-                "image": WINDOWS_LATEST["image_path"],
-                "dv_size": WINDOWS_LATEST["dv_size"],
+                "image": WINDOWS_LATEST.get("image_path"),
+                "dv_size": WINDOWS_LATEST.get("dv_size"),
```
this is done for collection only; during test execution, tests will not be executed if not part of the architecture and have windows set in pyconfig

rnetser commented on tests/virt/node/migration_and_maintenance/test_post_copy_migration.py:
```diff
@@ -104,12 +103,12 @@ def drained_node_with_hotplugged_vm(admin_client, hotplugged_vm):
             {
                 "dv_name": "dv-windows-latest-vm",
-                "image": os.path.join(Images.Windows.DIR, Images.Windows.WIN11_IMG),
+                "image": f"{Images.Windows.DIR}/{Images.Windows.WIN11_IMG}",
```
this is done for collection only; during test execution, tests will not be executed if not part of the architecture and have windows set in pyconfig

SiboWang1997 commented on utilities/constants.py:
```diff
@@ -26,96 +26,132 @@
+
+        Rhel = Rhel()
+        Rhel.RHEL9_5_IMG = "rhel-95-aarch64.qcow2"
+        Rhel.LATEST_RELEASE_STR = Rhel.RHEL9_5_IMG
```
OK, I will update #860 after this PR merged

vsibirsk commented on tests/global_config_arm64.py:
```diff
@@ -47,30 +39,18 @@
-    },
-]
+rhel_os_matrix = generate_os_matrix_dict(os_name="rhel", supported_operating_systems=["rhel-9-5"])
+fedora_os_matrix = generate_os_matrix_dict(os_name="fedora", supported_operating_systems=["fedora-41"])
+centos_os_matrix = generate_os_matrix_dict(os_name="centos", supported_operating_systems=["centos-stream-9"])
```
we don't have fedora and/or centos arm images

vsibirsk commented on utilities/constants.py:
```diff
@@ -26,96 +26,132 @@
+        Rhel.LATEST_RELEASE_STR = Rhel.RHEL9_5_IMG
+
+        Fedora = Fedora()
+        Fedora.FEDORA41_IMG = "Fedora-Cloud-Base-Generic-41-1.4.x86_64.qcow2"
```
it's x86 image. we don't have arm fedora image rn

dshchedr reviewed (changes requested):
nit

rnetser commented:
/build-and-push-container

openshift-virtualization-qe-bot-2 commented:
New container for quay.io/openshift-cnv/openshift-virtualization-tests:pr-983 published

rnetser commented on utilities/virt.py:
```diff
@@ -1550,27 +1550,33 @@ def get_guest_os_info(vmi):
-    ]
-    if windows_os_dict:
-        return windows_os_dict[0]
-    raise KeyError(f"Failed to extract {windows_version} from system_windows_os_matrix")
+    if system_windows_os_matrix := py_config.get("system_windows_os_matrix"):
+        windows_os_dict = [
+            os_dict
+            for win_os in system_windows_os_matrix
+            for os_name, os_dict in win_os.items()
+            if os_name == windows_version
+        ]
+        if windows_os_dict:
+            return windows_os_dict[0]
+        raise KeyError(f"Failed to extract {windows_version} from system_windows_os_matrix")
+
+    return {}
```
done

rnetser commented on utilities/virt.py:
```diff
@@ -1550,27 +1550,33 @@ def get_guest_os_info(vmi):
-    ]
-    if rhel_os_dict:
-        return rhel_os_dict[0]
-    raise KeyError(f"Failed to extract {rhel_version} from system_rhel_os_matrix")
+    if py_system_rhel_os_matrix := py_config.get("system_rhel_os_matrix"):
+        rhel_os_dict = [
+            os_dict
+            for rhel_os in py_system_rhel_os_matrix
+            for os_name, os_dict in rhel_os.items()
+            if os_name == rhel_version
+        ]
+        if rhel_os_dict:
+            return rhel_os_dict[0]
+        raise KeyError(f"Failed to extract {rhel_version} from system_rhel_os_matrix")
+
+    return {}
```
done

rnetser commented:
/build-and-push-container

openshift-virtualization-qe-bot-2 commented:
New container for quay.io/openshift-cnv/openshift-virtualization-tests:pr-983 published

hmeir commented on tox.ini:
```diff
@@ -16,16 +16,44 @@ setenv =
+deps=
+    uv
+commands =
+    uv run pytest --tc-file=tests/global_config_arm64.py --collect-only -m arm64
```
What about smoke tests and setupplan?

hmeir commented on tests/storage/cdi_import/test_import_http.py:
```diff
@@ -76,6 +76,8 @@
     "configmap_name": INTERNAL_HTTP_CONFIGMAP_NAME,
 }
 
+LATEST_WINDOWS_OS_DICT = py_config.get("latest_windows_os_dict", {})
```
Isn't this const redudant?
We have `WINDOWS_LATEST` at tests/os_params

rnetser replied:
this pr is huge; we can fix that in a separate PR> this PR is only for the addition of archs (if it was a small PR, I would do it)

hmeir commented on tests/virt/node/hotplug/test_cpu_memory_hotplug.py:
```diff
@@ -33,6 +32,8 @@
 LOGGER = logging.getLogger(__name__)
 TESTS_CLASS_NAME = "TestCPUHotPlug"
 
+LATEST_WINDOWS_OS_DICT = py_config.get("latest_windows_os_dict", {})
```
ditto

rnetser replied:
ditto

hmeir commented on tests/virt/node/hotplug/test_cpu_memory_hotplug.py:
```diff
@@ -142,12 +143,12 @@ def test_hotplug_cpu_above_max_value(self, hotplugged_vm):
             },
             {
-                "template_labels": py_config["latest_windows_os_dict"]["template_labels"],
+                "template_labels": py_config.get("latest_windows_os_dict", {}).get("template_labels"),
```
Shold use the const

hmeir commented on tests/virt/node/migration_and_maintenance/test_post_copy_migration.py:
```diff
@@ -104,12 +103,12 @@ def drained_node_with_hotplugged_vm(admin_client, hotplugged_vm):
             },
             {
-                "template_labels": py_config["latest_windows_os_dict"]["template_labels"],
+                "template_labels": py_config.get("latest_windows_os_dict", {}).get("template_labels"),
```
ditto

hmeir commented on tests/storage/hpp/test_hostpath.py:
```diff
@@ -595,7 +595,7 @@ def test_hostpath_registry_import_dv(
             {
                 "dv_name": "cnv-3516-source-dv",
-                "image": py_config["latest_fedora_os_dict"]["image_path"],
+                "image": py_config.get("latest_fedora_os_dict", {}).get("image_path"),
```
Should use FEDORA_LATEST.get("image_path")

rnetser replied:
my comment on the PR size and contents applies to all related comments

hmeir commented on tests/storage/test_data_protection.py:
```diff
@@ -69,8 +69,8 @@ def test_remove_cdi_pvc(skip_test_if_no_hpp_sc, pvc_hpp, cdi):
                 "dv_name": "cnv-3650",
-                "image": py_config["latest_fedora_os_dict"]["image_path"],
-                "dv_size": py_config["latest_fedora_os_dict"]["dv_size"],
+                "image": py_config.get("latest_fedora_os_dict", {}).get("image_path"),
+                "dv_size": py_config.get("latest_fedora_os_dict", {}).get("dv_size"),
```
ditto

hmeir commented on tests/infrastructure/workload_availability/remediation_fencing_mhc/test_ha_vm.py:
```diff
@@ -24,9 +24,9 @@
-    "dv_name": py_config["latest_fedora_os_dict"]["template_labels"]["os"],
-    "image": py_config["latest_fedora_os_dict"]["image_path"],
-    "dv_size": py_config["latest_fedora_os_dict"]["dv_size"],
+    "dv_name": py_config.get("latest_fedora_os_dict", {}).get("template_labels", {}).get("os"),
+    "image": py_config.get("latest_fedora_os_dict", {}).get("image_path"),
+    "dv_size": py_config.get("latest_fedora_os_dict", {}).get("dv_size"),
```
ditto

openshift-virtualization-qe-bot-3 commented:
New container for quay.io/openshift-cnv/openshift-virtualization-tests:pr-983 published

rnetser commented on tests/storage/cdi_upload/test_upload_virtctl.py:
```diff
@@ -40,6 +40,8 @@
 POPULATED_STR = "populated"
 NON_CSI_POPULATED_STR = "imported/cloned/updated"
 
+LATEST_WINDOWS_OS_DICT = py_config.get("latest_windows_os_dict", {})
```
ditto

rnetser commented on tests/storage/storage_migration/test_mtc_storage_class_migration.py:
```diff
@@ -84,9 +84,9 @@ def test_migrate_vms_after_storage_migration(self, booted_vms_for_storage_class_
+                "image": FEDORA_LATEST.get("image_path"),
                 "storage_class": py_config[STORAGE_CLASS_B],
-                "dv_size": FEDORA_LATEST["dv_size"],
+                "dv_size": FEDORA_LATEST.get("dv_size"),
```
this is for colletion only, we shoul fail in runtime if the value is not set

rnetser commented on tests/storage/storage_migration/test_mtc_storage_class_migration.py:
```diff
@@ -84,9 +84,9 @@ def test_migrate_vms_after_storage_migration(self, booted_vms_for_storage_class_
             {
                 "dv_name": "standalone-dv-fedora",
-                "image": FEDORA_LATEST["image_path"],
+                "image": FEDORA_LATEST.get("image_path"),
```
this is for colletion only, we shoul fail in runtime if the value is not set

rnetser commented:
/build-and-push-container

openshift-virtualization-qe-bot-3 commented:
New container for quay.io/openshift-cnv/openshift-virtualization-tests:pr-983 published

hmeir commented on pytest.ini:
```diff
@@ -87,7 +89,6 @@ addopts =
     -p no:logging
     --basetemp=/tmp/pytest
     --strict-markers
-    --tc-file=tests/global_config.py
```
Make sense

And what about the jenkins jobs?
Do they set --tc-file explicity? because if they dont, 4.19 amd jobs may break

rnetser replied:
yes, there's a jira for them to work on this. this PR cannot be merged before they make the change

rnetser commented:
/build-and-push-container

rnetser commented on utilities/os_utils.py:
```diff
@@ -0,0 +1,250 @@
+    Raises:
+        ValueError: If the OS name is not supported or if the supported operating systems list is empty.
+    """
+    match os_name:
+        case "rhel":
+            base_dict = RHEL_OS_MAPPING
+
+        case "windows":
+            base_dict = WINDOWS_OS_MAPPING
+
+        case "fedora":
+            base_dict = FEDORA_OS_MAPPING
+
+        case "centos":
+            base_dict = CENTOS_OS_MAPPING
+
+        case _:
```
done

rnetser commented on utilities/constants.py:
```diff
@@ -100,22 +101,104 @@ class Fedora:
+            CENTOS_STREAM_9_IMG = "CentOS-Stream-GenericCloud-9-20220107.0.x86_64.qcow2"
+            DIR = f"{BASE_IMAGES_DIR}/centos-images"
+            DEFAULT_DV_SIZE = "15Gi"
+            LATEST_RELEASE_STR = CENTOS_STREAM_9_IMG
```
removed rhel and fedora

rnetser commented:
> @rnetser Is my understanding correct going forward **every test will have at-least one architecture marker** (arm64, x86_64, s390x)?
> 
> If answer to my question is yes, I'm wondering why we've removed arm64 marker here https://github.com/RedHatQE/openshift-virtualization-tests/pull/983/files#diff-52ae305b40e049dcf643643010ad6d8bea60f51746421201c3242956f42d7015L81

Thanks for catching this, removed by mistake

openshift-virtualization-qe-bot-2 commented:
New container for quay.io/openshift-cnv/openshift-virtualization-tests:pr-983 published

rnetser commented on utilities/os_utils.py:
```diff
@@ -0,0 +1,244 @@
+    if not base_dict:
+        raise ValueError(f"Unsupported OS: {os_name}. Supported: rhel, windows, fedora, centos")
+
+    os_base_class = getattr(Images, os_name.title(), None)
+    if not os_base_class:
+        raise ValueError(
+            f"Unsupported OS: {os_name}. "
+            "Make sure it is supported under `utilities.constants.ArchImages` class for cluster architecture."
+        )
```
class name was changed to  "Centos"

rnetser commented:
/build-and-push-container

dshchedr commented on utilities/constants.py:
```diff
@@ -26,96 +26,132 @@
-BASE_IMAGES_DIR = "cnv-tests"
 NON_EXISTS_IMAGE = "non-exists-image-test-cnao-alerts"
 
+AMD_64 = "amd64"
```
thanks for looking on it!

geetikakay commented on libs/infra/images.py:
```diff
@@ -0,0 +1,97 @@
+    RHEL9_4_IMG: str | None = None
+    RHEL9_5_IMG: str | None = None
+    RHEL9_6_IMG: str | None = None
+    RHEL8_REGISTRY_GUEST_IMG: str = "registry.redhat.io/rhel8/rhel-guest-image"
```
use Resource.ApiGroup.IMAGE_REGISTRY

rnetser replied:
done

geetikakay commented on tests/global_config_s390x.py:
```diff
@@ -0,0 +1,35 @@
+from typing import Any
```
I think it can be skipped ? we branched out s390x branch

rnetser commented:
replaced by https://github.com/RedHatQE/openshift-virtualization-tests/pull/1244

openshift-virtualization-qe-bot-2 commented:
Successfully removed PR tag: quay.io/openshift-cnv/openshift-virtualization-tests:pr-983.
